src/server/index.tsx

```diff
@@ -48,9 +49,66 @@ export const createServer = ({
   rootPath,
   _injectAuthenticationToken = injectAuthenticationToken,
 }: CreateServerParams) => {
+  // This CSP is a tight default. This CSP should be on all requests
+  // that aren't serving a document. This CSP should be set with statics.
+  const baseCSP = {
+    defaultSrc: ["'none'"],
+    baseUri: ["'none'"],
+    formAction: ["'none'"],
+    objectSrc: ["'none'"],
+    reportUri: '/__cspreport__',
+  };
+
+  // This config sets the non-static CSP for deployed instances.
+  // TODO: A separate CDN should be configured for statics. When that happens
+  // the 'self' should be removed and replaced with the CDN host + path for statics.
```

Comment:
Could you file an issue for that please?

Reply:
Yep I'll file any follow-up issues after this lands.